### PR TITLE
Split vesting schedule, added sealing, variable initial token amounts

### PR DESCRIPTION
**Pull request by thekevinbrown**
**Vesting Schedule**
The previous vesting schedule was capped in size because it was all in one array. Given how many we're likely to have, this PR splits the amounts into a mapping and allows multiple separate calls to initialise the full schedule.

As such, there needs to be a way to indicate we're done with that. I've added sealing, and once sealed the contract cannot be unsealed, even by the owner.

**Token**
Pulled initial amounts out into the initialize function so we can choose them later.

### CODE REVIEW

File: contracts/Vesting.sol [hunk 1]
```diff
@@ -1,62 +1,88 @@
...
+        require(newEntries.length > 0, "Vesting: no entries");
+
+        for (uint i = 0; i < newEntries.length; i++) {
+            schedule[to].push(newEntries[i]);
         }
+        
+        emit ScheduleChanged(to, schedule[to]);
     }
 
+    event ScheduleChanged(address to, VestingSchedule[] newSchedule);
```
Comment (bitcoinbrisbane) [inline contracts/Vesting.sol:39]: I like to index

```suggestion
    event ScheduleChanged(address indexed to, VestingSchedule[] newSchedule);
```

**Review by bitcoinbrisbane — COMMENTED** [review]
One suggestion

File: contracts/Vesting.sol (hunk 1 above)
Comment (thekevinbrown) [reply contracts/Vesting.sol:39]: 👍

**Review by bitcoinbrisbane — APPROVED** [review]
LGTM

**Pull request merged**